vortex-array/src/array/byte_bool/compute/mod.rs

```diff
@@ -0,0 +1,242 @@
+use std::ops::{BitAnd, BitOr, BitXor, Not};
+use std::sync::Arc;
+
+use arrow_array::{ArrayRef as ArrowArrayRef, BooleanArray as ArrowBoolArray};
+use arrow_buffer::{BooleanBuffer, BooleanBufferBuilder};
+use num_traits::AsPrimitive;
+use vortex_dtype::{match_each_integer_ptype, Nullability};
+use vortex_error::{vortex_bail, VortexResult};
+use vortex_expr::Operator;
+use vortex_scalar::Scalar;
+
+use super::{ByteBoolArray, ByteBoolMetadata};
+use crate::ToArrayData;
+use crate::{
+    compute::{
+        as_arrow::AsArrowArray, compare::CompareFn, fill::FillForwardFn, scalar_at::ScalarAtFn,
+        slice::SliceFn, take::TakeFn, ArrayCompute,
+    },
+    encoding::ArrayEncodingRef,
+    stats::StatsSet,
+    validity::ArrayValidity,
+    ArrayDType, ArrayData, ArrayTrait, IntoArray,
+};
+
+impl ArrayCompute for ByteBoolArray {
+    fn as_arrow(&self) -> Option<&dyn crate::compute::as_arrow::AsArrowArray> {
+        Some(self)
+    }
+
+    fn compare(&self) -> Option<&dyn crate::compute::compare::CompareFn> {
+        Some(self)
+    }
+
+    fn fill_forward(&self) -> Option<&dyn crate::compute::fill::FillForwardFn> {
+        None
+    }
+
+    fn scalar_at(&self) -> Option<&dyn ScalarAtFn> {
+        Some(self)
+    }
+
+    fn slice(&self) -> Option<&dyn crate::compute::slice::SliceFn> {
+        Some(self)
+    }
+
+    fn take(&self) -> Option<&dyn crate::compute::take::TakeFn> {
+        Some(self)
+    }
+}
+
+impl ScalarAtFn for ByteBoolArray {
+    fn scalar_at(&self, index: usize) -> VortexResult<Scalar> {
+        if index >= self.len() {
+            vortex_bail!(OutOfBounds: index, 0, self.len());
+        }
+
+        let scalar = match self.is_valid(index).then(|| self.buffer()[index] == 1) {
+            Some(b) => Scalar::new(
+                vortex_dtype::DType::Bool(self.validity().nullability()),
+                vortex_scalar::ScalarValue::Bool(b),
+            ),
+            None => Scalar::null(self.dtype().clone()),
+        };
+
+        Ok(scalar)
+    }
+}
+
+impl AsArrowArray for ByteBoolArray {
+    fn as_arrow(&self) -> VortexResult<ArrowArrayRef> {
+        let nulls = self.logical_validity().to_null_buffer()?;
+
+        let mut builder = BooleanBufferBuilder::new(self.len());
+        builder.append_slice(self.as_ref());
+
+        Ok(Arc::new(ArrowBoolArray::new(builder.finish(), nulls)))
+    }
+}
+
+impl SliceFn for ByteBoolArray {
+    fn slice(&self, start: usize, stop: usize) -> VortexResult<crate::Array> {
+        let length = stop - start;
+
+        let validity = self.validity().slice(start, stop)?;
+
+        let slice_metadata = Arc::new(ByteBoolMetadata {
+            validity: validity.to_metadata(length)?,
```

Comment:
Just a note, this eagerly re-computes whether the slice is AllValid/AllInvalid. Which is nice, just maybe leave a comment since it's perhaps not obvious why this is done

Reply:
Not sure I understand, seems like it just keeps the existing validity, with slight adjustment for arrays.

Reply:
Not sure what Nick meant here, it's `to_logical` that does the evaluation